main_test.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+var a Job
+
+func TestMain(m *testing.M) {
+
+	a.Initialize("DB_USERNAME", "DB_PASSWORD", "rest_api_example")
+	ensureTableExists()
+	code := m.Run()
+	clearTable()
+	os.Exit(code)
+}
+func ensureTableExists() {
+	if _, err := a.DB.Exec(tableCreationQuery); err != nil {
+		log.Fatal(err)
```

Comment:
instead of `log.Fatal(err)` , since we're going to be receiving `m` as a parameter, we wanna call `m.Errorf("table doesn't exist")`

Reply:
https://golang.org/pkg/testing/